lib/tld.js

```diff
@@ -141,7 +141,7 @@ tld.prototype.getRulesForTld = function getRulesForTld (tld, default_rule) {
   }
 
   // Nothing found, apply some default value
-  if (!rules) {
+  if (rules === void 0) {
```

Comment:
As far as I can tell, `rules` is a string (based on looking at https://github.com/oncletom/tld.js/blob/f1477fbae4d60c1c2eb3b5dfc91e969dd7eb00a0/rules.json). It can be `""`, `"*"`, or a longer string that is `|` delimited.

If it's `undefined`, it means that the requested TLD does not exist in rules.json, thus is not a known TLD, so it should have a default rule applied.

In the existing code, trying to find a public suffix on any TLD that is defined to have a rule of `""` will fail because the default rule is applied. But something like `.eu` is a valid domain, so we don't want default rules applied.

Or am I misunderstanding, and `""` means we should apply a default rule? I'm currently reading it as "apply the default rule only if the TLD is unknown"